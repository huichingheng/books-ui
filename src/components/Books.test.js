import React from "react";
import Books from "./Books";
import ShallowRender from "react-test-renderer/shallow";


test("renders Book component", () => {
  const renderer = new ShallowRender();
  renderer.render(<Books />);
  const output = renderer.getRenderOutput();

  expect(output).toMatchSnapshot();
});

// test before the component mount
test("renders Book title", () => {
  const renderer = new ShallowRender();
  renderer.render(<Books />);
  const output = renderer.getRenderOutput();

  expect(output).toMatchSnapshot();
});

test("renders title and booklist after the component mount", async ()=>{
  fetch.mockResponseOnce(
    JSON.stringify([
      {
        title: "Harry Potter",
        _id: "654723849s"
      },
      {
        title: "John Wick",
        _id: "528940562j"
      },
    ])
  )
  const renderer = new ShallowRender();
  renderer.render(<Books />)

  const instance = renderer.getMountedInstance()
  await instance.componentDidMount()

  const output = renderer.getRenderOutput();
  expect(output).toMatchSnapshot();
})