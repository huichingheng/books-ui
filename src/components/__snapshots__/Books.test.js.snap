// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders Book component 1`] = `
<div>
  <h2>
    Books
  </h2>
  ;
</div>
`;

exports[`renders Book title 1`] = `
<div>
  <h2>
    Books
  </h2>
  ;
</div>
`;

exports[`renders title and booklist after the component mount 1`] = `
<div>
  <h2>
    Books
  </h2>
  <li>
    Harry Potter
  </li>
  <li>
    John Wick
  </li>
  ;
</div>
`;
